components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/core/CarbonAxisConfigurator.java

```diff
@@ -484,4 +482,20 @@ private InputStream getAxis2XmlInputStream() throws AxisFault {
         return axis2xmlStream;
     }
 
+    /**
+     * Deploy all services in the given repoLocation.
+     */
+    public void deployServices() {
+        if (repoLocation != null && repoLocation.trim().length() != 0) {
```

Comment:
The repoLocation cannot be null/empty when the code reaches here. It should fail at the init method. WDYT?

Reply:
Yes. If the repoLocation is null, WSO2 Carbon initialization will be failed from the init() method by throwing a ServerException. Thereafter, deployServices() method will not get invoked. Therefore, not need to have this null check at this point. Corrected.